lib/Service/FolderService.php

```diff
@@ -226,19 +230,21 @@ public function deleteShare($shareId): void {
 	 * @param int $folderId
 	 * @param string $title
 	 * @param int $parent_folder
+	 * @param int $perms
 	 * @return Folder|SharedFolder
 	 * @throws DoesNotExistException
 	 * @throws MultipleObjectsReturnedException
+	 * @throws UnauthorizedAccessError
 	 * @throws UnsupportedOperation
 	 * @throws \OCA\Bookmarks\Exception\UrlParseError
 	 */
-	public function updateSharedFolderOrFolder($userId, $folderId, $title = null, $parent_folder = null) {
+	public function updateSharedFolderOrFolder($userId, $folderId, $title = null, $parent_folder = null, $perms = Authorizer::PERM_NONE) {
 		/**
 		 * @var $folder Folder
 		 */
 		$folder = $this->folderMapper->find($folderId);
 
-		if ($userId !== null || $userId !== $folder->getUserId()) {
+		if ($userId === null || $userId !== $folder->getUserId()) {
```

Comment:
Any reason for this change? I'm not convinced by the original line but neither by this new version. $userId shouldn't be null here, I believe because we need it to find the sharedFolder. Maybe my suggestion makes sense?

```suggestion
		if ($userId !== null && $userId !== $folder->getUserId()) {
```

Reply:
I just came across this line and thought it is suspicious. I guess it works, because there will be no shares for the owner and so the code doesn't do anything in that case. Your version makes sense, if there is no way for unregistered users to edit folders.